engine/table/src/main/java/io/deephaven/engine/table/impl/FilterContext.java

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.impl.select.WhereFilter;
+import io.deephaven.util.SafeCloseable;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Helper class to manage filters and their associated execution cost.
+ */
+public class FilterContext implements SafeCloseable {
+    private final WhereFilter filter;
+    private final PushdownPredicateManager ppm;
+    private final AbstractColumnSource<?> columnSource;
+
+    // Store results of the push-down operation.
+    private PushdownResult pushdownResult = null;
+
+    // As we execute potentially many push down filters, store the most recent (and maximum) cost experienced so far.
+    private long executedFilterCost;
+    // The cost of executing the next pushdown filter step.
+    private long pushdownFilterCost;
+
+    protected FilterContext(
+            WhereFilter filter,
+            PushdownPredicateManager ppm,
+            AbstractColumnSource<?> columnSource) {
+        this.filter = filter;
+        this.ppm = ppm;
+        this.columnSource = columnSource;
+        executedFilterCost = 0;
+        pushdownFilterCost = Long.MAX_VALUE;
+    }
+
+    public static FilterContext of(WhereFilter filter, PushdownPredicateManager ppm) {
+        return new FilterContext(filter, ppm, null);
+    }
+
+    public static FilterContext of(WhereFilter filter, AbstractColumnSource<?> columnSource) {
+        return new FilterContext(filter, null, columnSource);
+    }
+
+    public static FilterContext of(WhereFilter filter) {
+        return new FilterContext(filter, null, null);
+    }
+
+    public WhereFilter filter() {
+        return filter;
+    }
+
+    public PushdownResult pushdownResult() {
+        return pushdownResult;
+    }
+
+    public void updatePushdownResult(final PushdownResult result) {
+        if (pushdownResult != null) {
+            pushdownResult.close();
+        }
+        this.pushdownResult = result;
+    }
+
+    public PushdownPredicateManager ppm() {
+        return ppm;
+    }
+
+    public AbstractColumnSource<?> columnSource() {
+        return columnSource;
+    }
+
+    public long executedFilterCost() {
+        return executedFilterCost;
+    }
+
+    public void updateExecutedFilterCost(final long cost) {
```

Comment:
We should have Javadoc on these methods, so that as you navigate the other code you can see what the meaning of the methods is.